### PR TITLE
Add firebase auth support to endpoints sample.

### PR DESCRIPTION
**Pull request by jeffmendoza**
I couldn't find any example clients or docs for testing firebase auth in the Python or Go examples, so going with this until we have more docs on https://cloud.google.com/endpoints/docs/authenticating-users

### CODE REVIEW

**Comment by jeffmendoza** [comment]
Fixes #32

**Comment by jeffmendoza** [comment]
logging and language are failing, appengine/endpoints passes:
```
 bundle exec rspec
Randomized with seed 2282
Ruby Endpoints Sample
  handles Firebase Auth
  POST /echo renders message from request body
  GET /auth/info/auth0 renders user info
  handles exceptions by returning Swagger-compliant JSON
  GET /auth/info/googlejwt renders user info
  GET /auth/info/googleidtoken renders user info
Finished in 0.04967 seconds (files took 0.14035 seconds to load)
6 examples, 0 failures
Randomized with seed 2282
```
cc @remi @frankyn

**Comment by remi** [comment]
Awesome!  Taking a look...

File: appengine/endpoints/app.rb [hunk 1]
```diff
@@ -20,11 +20,21 @@
 require "base64"
 require "json"
 require "sinatra"
+require "rack/cors"
 
 before do
   content_type :json
 end
 
+use Rack::Cors do
```
Comment (remi) [inline appengine/endpoints/app.rb]: Nice find with Rack::Cors 👍

File: appengine/endpoints/app.rb [hunk 2]
```diff
@@ -56,6 +66,19 @@ def auth_info
   auth_info
 end
 
+# Auth info with Firebase Auth.
+get "/auth/info/firebase" do
+  auth_info
+end
+options "/auth/info/firebase" do
```
Comment (remi) [inline appengine/endpoints/app.rb]: nit: can you add a newline above `options "..." do` so the handlers are separated by 1 newline?

File: appengine/endpoints/app.rb (hunk 1 above)
Comment (jeffmendoza) [inline appengine/endpoints/app.rb]: Aja pointed me to: https://til.hashrocket.com/posts/4d7f12b213-rails-5-api-and-cors

File: appengine/endpoints/app.rb (hunk 2 above)
Comment (jeffmendoza) [inline appengine/endpoints/app.rb]: updated

**Comment by jeffmendoza** [comment]
Updated

**Pull request merged**